src/commands/utils/helper/kubectlGadgetDownload.ts

```diff
@@ -0,0 +1,62 @@
+import * as vscode from 'vscode';
+import * as os from 'os';
+import { getKubectlGadgetConfig } from '../config';
+import { Errorable, failed } from '../errorable';
+import { getToolBinaryPath } from './binaryDownloadHelper';
+
+async function getLatestKubectlGadgetReleaseTag() {
+    const kubegadgetConfig = getKubectlGadgetConfig();
+    if (failed(kubegadgetConfig)) {
+        vscode.window.showErrorMessage(kubegadgetConfig.error);
+        return undefined;
+    }
+
+    return kubegadgetConfig.result.releaseTag;
+}
+
+export async function getKubectlGadgetBinaryPath(): Promise<Errorable<string>> {
+    const releaseTag = await getLatestKubectlGadgetReleaseTag();
+    if (!releaseTag) {
+         return {succeeded: false, error: "Could not get latest release tag."};
+    }
+
+    const archiveFilename = getArchiveFilename(releaseTag);
+    const downloadUrl = `https://github.com/inspektor-gadget/inspektor-gadget/releases/download/${releaseTag}/${archiveFilename}`;
+    const pathToBinaryInArchive = getPathToBinaryInArchive();
```

Comment:
I would suggest to not use a fixed version here and instead download always the latest. Otherwise, we will have to create a PR every time we release a new version (consider we do it monthly). 

We could use the GitHub API to retrieve the latest version (`tag_name` JSON field): [here](https://api.github.com/repos/inspektor-gadget/inspektor-gadget/releases/latest). And then, compose the URL for the `kubectl-gadget` binary, as you are doing right now. 

Of course, I would suggest to verify if we are using the latest version only for the `Deploy` operation, and not every time we run a new gadget.

Reply:
@blanquicet: we generally tie the VSCode extension to specific versions of all our dependencies, to make its behaviour as predictable as possible (although we allow users to override this in settings.json, which is mainly helpful for development purposes). We do quite frequent releases of the VSCode extension too, and it's trivial for us to bump the versions of our dependencies. If that does become too onerous, we could consider generating automated PRs to bump the IG version.

On a sort-of related note, this reminds me of another question I had (probably because I haven't read the docs properly, so feel free to point me there): if you run `gadget deploy` with one version, and then `gadget undeploy` with a later version, is that guaranteed to work? Or does it lead to the possibility of errors or orphaned resources?

I wouldn't be surprised if there will be circumstances in the future where we'd want to check whether IG is installed on the cluster, and perhaps the version (maybe we'd want to selectively enable/disable the deploy and undeploy commands, or give users the option to upgrade). Is there a recommended way of checking this?

Reply:
> We do quite frequent releases of the VSCode extension too, and it's trivial for us to bump the versions of our dependencies. 
>
Ahh ok, great to know.

> If that does become too onerous, we could consider generating automated PRs to bump the IG version.
>
That would be great!

> if you run gadget deploy with one version, and then gadget undeploy with a later version, is that guaranteed to work? Or does it lead to the possibility of errors or orphaned resources? 
>
Yep, it is. There was only one version it wasn't because we changed the namespace where Inspektor Gadget was running. We notified it in the release notes and also in also slack. So, even if I don't expect to have another change that doesn't guarantee this, it will be for sure specified in the release notes, and we can decide how to behave in that case. Is it OK for you?

@mauriciovasquezbernal any comment?

> I wouldn't be surprised if there will be circumstances in the future where we'd want to check whether IG is installed on the cluster, and perhaps the version (maybe we'd want to selectively enable/disable the deploy and undeploy commands, or give users the option to upgrade). Is there a recommended way of checking this?
>
We could use the `version` command (NOTE: Maybe we could add a JSON output mode for that `version` command so that it is easier to parse the server version):

```bash
$ kubectl gadget version
Client version: v0.12.0
Server version: v0.12.0
```

But I would suggest to work on this only if we release a version that doesn't guarantee backward compatibility.